File: src/main/java/net/atos/zac/zoeken/model/ZoekResultaat.java
package net.atos.zac.zoeken.model;

import java.util.EnumMap;
import java.util.List;

import net.atos.zac.shared.model.Resultaat;

public class ZoekResultaat<TYPE> extends Resultaat<TYPE> {

    private final EnumMap<FilterVeld, List<FilterResultaat>> filters = new EnumMap<>(FilterVeld.class);

    public ZoekResultaat(final List<TYPE> items, final long count) {
        super(items, count);
    }

    public EnumMap<FilterVeld, List<FilterResultaat>> getFilters() {
        return filters;
    }

    public void addFilter(final FilterVeld facetVeld, final List<FilterResultaat> waardes) {
        filters.put(facetVeld, waardes);
    }
}

File: src/main/app/src/app/zoeken/model/zoek-resultaat.ts
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

import { Resultaat } from "../../shared/model/resultaat";
import { FilterVeld } from "./filter-veld";
import { FilterResultaat } from "./filter-resultaat";

export class ZoekResultaat<TYPE> extends Resultaat<TYPE> {
  filters: Partial<Record<FilterVeld, FilterResultaat[]>> = {};
}
